src/form/form-password-input.tsx

```diff
@@ -1,19 +1,54 @@
 import { useField } from 'formik';
-import React, { FC } from 'react';
+import React, { FC, useState } from 'react';
 
+import { Divider } from '../components/divider/divider';
+import { PasswordStrengthIndicator } from '../components/password-strength-indicator/password-strength-indicator';
 import { StyledPasswordInput } from '../components/styled-password-input/styled-password-input';
 import { StyledPasswordInputProps } from '../components/styled-password-input/styled-password-input.props';
+import { formatSize } from '../styles/format-size';
 import { hasError } from '../utils/has-error';
+import { isDefined } from '../utils/is-defined';
 import { ErrorMessage } from './error-message/error-message';
 
+const MIN_PASSWORD_LENGTH = 8;
+const uppercaseLowercaseMixtureRegx = /(?=.*[a-z])(?=.*[A-Z])/;
+const lettersNumbersMixtureRegx = /(?=.*\d)(?=.*[A-Za-z])/;
+const specialCharacterRegx = /[!@#$%^&*()_+\-=\]{};':"\\|,.<>?]/;
+
+export interface PasswordValidation {
+  minChar: boolean;
+  cases: boolean;
+  number: boolean;
+  specialChar: boolean;
+}
+
 interface Props extends Pick<StyledPasswordInputProps, 'testID'> {
   name: string;
+  isNewPassword?: boolean;
 }
 
-export const FormPasswordInput: FC<Props> = ({ name, testID }) => {
+export const FormPasswordInput: FC<Props> = ({ name, isNewPassword, testID }) => {
   const [field, meta, helpers] = useField<string>(name);
   const isError = hasError(meta);
 
+  const [focused, setFocused] = useState(false);
+  const [passwordValidation, setPasswordValidation] = useState<PasswordValidation>({
+    minChar: false,
+    cases: false,
+    number: false,
+    specialChar: false
+  });
+
+  const handleChange = (text: string) => {
+    helpers.setValue(text);
+    setPasswordValidation({
+      minChar: text.length >= MIN_PASSWORD_LENGTH,
+      cases: uppercaseLowercaseMixtureRegx.test(text),
+      number: lettersNumbersMixtureRegx.test(text),
+      specialChar: specialCharacterRegx.test(text)
+    });
+  };
```

Comment:
The same mistake that was done on extension.  
This functionality is not related to shared password input.  extract this logic out of here.